WordPress/gutenberg
# Post fields: move `date` fields from `edit-site` to `fields` package

Part of https://github.com/WordPress/gutenberg/issues/61084

## What?

This PR moves the `date` field definition from `edit-site` to the `fields` package.

## Why?

In preparation for moving the whole `usePostFields` hook ([conversation](https://github.com/WordPress/gutenberg/pull/65196#discussion_r1777506405)).

## Testing Instructions

Visit the Pages page and verify that the "Date" field still works as expected (visualization, edit).

<img width="1533" alt="Screenshot 2024-11-12 at 16 44 45" src="https://github.com/user-attachments/assets/bdb31ea1-3939-4cf2-ab38-c2c9d1280072">
